src/registrar/admin.py

```diff
@@ -1812,10 +1813,90 @@ def changelist_view(self, request, extra_context=None):
         return response
 
     def change_view(self, request, object_id, form_url="", extra_context=None):
+        """Display restricted warning,
+        Setup the auditlog trail and pass it in extra context."""
         obj = self.get_object(request, object_id)
         self.display_restricted_warning(request, obj)
+
+        # Initialize variables for tracking status changes and filtered entries
+        filtered_entries = []
+
+        try:
+            # Retrieve and order audit log entries by timestamp in ascending order
+            audit_log_entries = LogEntry.objects.filter(object_id=object_id).order_by("timestamp")
+
+            # Process each log entry to filter based on the change criteria
+            for log_entry in audit_log_entries:
+                entry = self.process_log_entry(log_entry)
+                if entry:
+                    filtered_entries.append(entry)
+
+        except ObjectDoesNotExist as e:
+            logger.error(f"Object with object_id {object_id} does not exist: {e}")
+        except Exception as e:
+            logger.error(f"An error occurred during change_view: {e}")
+
+        # Reverse the filtered entries list to get newest to oldest order
+        filtered_entries.reverse()
```

Comment:
Probably. I had a "previous_status" var that would've gotten messed-up, but have since revised to set up the 'previous status' based on the reason that was changed. Good call.